node/src/components/contract_runtime.rs

```diff
@@ -395,39 +402,43 @@ impl ContractRuntime {
                 let engine_state = Arc::clone(&self.engine_state);
                 let metrics = Arc::clone(&self.metrics);
                 async move {
-                    let result = Self::get_trie_full(&engine_state, &metrics, trie_key);
+                    let result =
+                        Self::get_trie_full(&engine_state.lock().unwrap(), &metrics, trie_key);
                     trace!(?result, "get_trie_full response");
                     responder.respond(result).await
                 }
                 .ignore()
             }
             ContractRuntimeRequest::PutTrie {
                 trie_bytes,
-                responder,
+                responder: _,
             } => {
                 trace!(?trie_bytes, "put_trie request");
                 let engine_state = Arc::clone(&self.engine_state);
-                let metrics = Arc::clone(&self.metrics);
+                let _ = Arc::clone(&self.metrics);
                 async move {
                     let correlation_id = CorrelationId::new();
-                    let start = Instant::now();
-                    let result = engine_state.put_trie_and_find_missing_descendant_trie_keys(
-                        correlation_id,
-                        &*trie_bytes,
-                    );
+                    let _ = Instant::now();
+                    let _ = engine_state
+                        .lock()
+                        .unwrap()
+                        .put_trie_and_find_missing_descendant_trie_keys(
+                            correlation_id,
+                            &*trie_bytes,
+                        );
                     // PERF: this *could* be called only periodically.
-                    if let Err(lmdb_error) = engine_state.flush_environment() {
-                        fatal!(
-                            effect_builder,
-                            "error flushing lmdb environment {:?}",
-                            lmdb_error
-                        )
-                        .await;
-                    } else {
-                        metrics.put_trie.observe(start.elapsed().as_secs_f64());
-                        trace!(?result, "put_trie response");
-                        responder.respond(result).await
-                    }
+                    // if let Err(lmdb_error) = engine_state.lock().unwrap().flush_environment() {
+                    //     fatal!(
+                    //         effect_builder,
+                    //         "error flushing lmdb environment {:?}",
+                    //         lmdb_error
+                    //     )
+                    //     .await;
+                    // } else {
+                    //     metrics.put_trie.observe(start.elapsed().as_secs_f64());
+                    //     trace!(?result, "put_trie response");
+                    //     responder.respond(result).await
+                    // }
```

Comment:
Why is this commented out?

Reply:
For the sake of simplicity, there was an issue moving `engine state` into an async closure.
I though we could live w/o `flush_environment()` and w/o metrics, but now I see there's also the `responder.respond(result)` commented out.
Need to rethink this.